src/org/jitsi/impl/neomedia/codec/audio/AbstractFFmpegAudioCodec.java

```diff
@@ -151,6 +151,17 @@ protected synchronized void doOpen()
             if (channels == Format.NOT_SPECIFIED)
                 channels = 1;
             FFmpeg.avcodeccontext_set_channels(avctx, channels);
+            if (channels == 1)
+            {
+                //mono
+                FFmpeg.avcodeccontext_set_channel_layout(avctx, 0x4);
+            }
+            else if (channels == 2)
+            {
+                //stereo
+                FFmpeg.avcodeccontext_set_channel_layout(avctx, 0x3);
```

Comment:
This is quite magical and not documented in FFmpeg.java

Reply:
And is the lack of a call to avcodeccontext_set_channels() intentional?

Reply:
Not sure I follow on this one. Line 153?

Reply:
This was mainly used for MP3 recording. It sets FFmpeg‘s AVCodecContext::channel_layout member. 3 and 4 are some FFmpeg constants that I should have added in FFmpeg.java rather than using the plain value (which is AV_CH_LAYOUT_MONO=4, AV_CH_LAYOUT_STEREO=3).